components/brave_federated/config_utils.cc

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/config_utils.h"
+
+#include "base/files/file_util.h"
+#include "base/json/json_reader.h"
+#include "base/strings/string_number_conversions.h"
+#include "brave/components/brave_federated/features.h"
+
+namespace brave_federated {
+
+LearningServiceConfig::LearningServiceConfig() {
+  reconnect_policy_ = {.num_errors_to_ignore = 0,
+                       .initial_delay_ms = 10 * 1000,
+                       .multiply_factor = 2.0,
+                       .jitter_factor = 0.0,
+                       .maximum_backoff_ms = 60 * 10 * 1000,
+                       .always_use_initial_delay = true};
+  request_task_policy_ = {
+      .num_errors_to_ignore = 0,
+      .initial_delay_ms =
+          features::GetFederatedLearningUpdateCycleInSeconds() * 1000,
```

Comment:
Using base::TimeDelta we would not need `* 1000`, because we could just call `.InMilliseconds`. Please see other comments